### PR TITLE
services/horizon: Add verify-range docker image

### PR DESCRIPTION
**Pull request by bartekn**
<details>
  <summary>PR Checklist</summary>
  
### PR Structure

* [x] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [x] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [x] This PR's title starts with name of package that is most changed in the PR, ex.
  `services/friendbot`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [x] This PR adds tests for the most critical parts of the new functionality or fixes.
* [x] I've updated any docs ([developer docs](https://www.stellar.org/developers/reference/), `.md`
  files, etc... affected by this change). Take a look in the `docs` folder for a given service,
  like [this one](https://github.com/stellar/go/tree/master/services/horizon/internal/docs).

### Release planning

* [x] I've updated the relevant CHANGELOG ([here](services/horizon/CHANGELOG.md) for Horizon) if
  needed with deprecations, added features, breaking changes, and DB schema changes.
* [x] I've decided if this PR requires a new major/minor version according to
  [semver](https://semver.org/), or if it's mainly a patch change. The PR is targeted at the next
  release branch if it's not a patch change.
</details>

### What

This commit adds a docker image that allows running multiple instances of `horizon expingest verify-command` on a single machine or running it in [AWS Batch](https://aws.amazon.com/batch/).

### Why

To be able to verify ingestion pipeline faster we are going to use AWS Batch to start a fleet of instances to ingest a set of ledger ranges.

### Known limitations

* The image needs to be updated with a script that verifies if history has been ingested correctly.
* Would be great to add duration limit of a single ingestion ledger to `verify-range` command to find performance bugs.

### CODE REVIEW

File: services/horizon/docker/verify-range/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,18 @@
+FROM ubuntu:18.04
+
+MAINTAINER Bartek Nowotarski <bartek@stellar.org>
+
+ENV STELLAR_CORE_VERSION 12.2.0-1088-3f8a5356
```
Comment (ire-and-curses) [inline services/horizon/docker/verify-range/Dockerfile]: Presumably we'll update this when https://github.com/stellar/stellar-core/pull/2383 lands?

File: services/horizon/docker/verify-range/README.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+## Env variables
+
+### Running locally
+
+Name | Description
+-|- 
+`BRANCH` | Git branch to build (useful for testing PRs)
+`FROM` | First ledger of the range (must be checkpoint ledger)
+`TO` | Last ledger of the range (must be checkpoint ledger)
```
Comment (ire-and-curses) [inline services/horizon/docker/verify-range/README.md]: Use http://markdowntable.com/ for these (note: can't auto suggest for a range)
```suggestion
| Name     | Description                                           |
|----------|-------------------------------------------------------|
| `BRANCH` | Git branch to build (useful for testing PRs)          |
| `FROM`   | First ledger of the range (must be checkpoint ledger) |
| `TO`     | Last ledger of the range (must be checkpoint ledger)  |
```

File: services/horizon/docker/verify-range/README.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+### Running in AWS Batch
+
+Name | Description
+-|- 
+`BRANCH` | Git branch to build (useful for testing PRs)
+`BATCH_START_LEDGER` | First ledger of the AWS Batch Job, must be a checkpoint ledger or 1.
+`BATCH_SIZE` | Size of the batch, must be multiple of 64.
```
Comment (ire-and-curses) [inline services/horizon/docker/verify-range/README.md]:
```suggestion
| Name                 | Description                                                          |
|----------------------|----------------------------------------------------------------------|
| `BRANCH`             | Git branch to build (useful for testing PRs)                         |
| `BATCH_START_LEDGER` | First ledger of the AWS Batch Job, must be a checkpoint ledger or 1. |
| `BATCH_SIZE`         | Size of the batch, must be multiple of 64.                           |
```

File: services/horizon/docker/verify-range/README.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+When you start 10 jobs with `BATCH_START_LEDGER=63` and `BATCH_SIZE=64`
+it will run the following ranges:
+
+`AWS_BATCH_JOB_ARRAY_INDEX` | `FROM` | `TO`
+-|-|-
+0 | 63 | 127
+1 | 127 | 191
+2 | 191 | 255
+3 | 255 | 319
```
Comment (ire-and-curses) [inline services/horizon/docker/verify-range/README.md]:
```suggestion
| `AWS_BATCH_JOB_ARRAY_INDEX` | `FROM` | `TO` |
|-----------------------------|--------|------|
| 0                           | 63     | 127  |
| 1                           | 127    | 191  |
| 2                           | 191    | 255  |
| 3                           | 255    | 319  |
```

File: services/horizon/docker/verify-range/README.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+## Tips when using AWS Batch
+
+* In "Job definition" set vCPUs to 2 and Memory to 4096 what represents `c5.large` instances Horizon should be using.
```
Comment (ire-and-curses) [inline services/horizon/docker/verify-range/README.md]:
```suggestion
* In "Job definition" set vCPUs to 2 and Memory to 4096. This represents the `c5.large` instances Horizon should be using.
```

File: services/horizon/docker/verify-range/Dockerfile (hunk 1 above)
Comment (bartekn) [reply services/horizon/docker/verify-range/Dockerfile]: #2111 should fix it.

**Comment by bartekn** [comment]
@ire-and-curses thanks for markdown table suggestion. Actually a plugin that formats tables in my VS Code disabled itself after upgrade. Fixed.

Not merging it yet because I want to add history tables diff post https://github.com/stellar/go/issues/2109.

**Comment by tamirms** [comment]
would it make sense to move this to https://github.com/stellar/go/tree/master/exp/tools or https://github.com/stellar/go/tree/master/tools ?

**Comment by tamirms** [comment]
@bartekn 

> Not merging it yet because I want to add history tables diff post #2109.

one easy way to verify the history tables is to call the `CheckExp...` functions. You can add some configuration which controls how the check functions are invoked. During `verify-range` it probably makes sense for a `CheckExp...` failure to cause the pipeline to shutdown with an error

**Comment by bartekn** [comment]
> would it make sense to move this to https://github.com/stellar/go/tree/master/exp/tools or https://github.com/stellar/go/tree/master/tools ?

I wonder if we should create `/docker` (in root) where we can keep all of the docker images connected to this repo.

**Pull request merged**